Data/EnumSet.hs

```diff
@@ -119,8 +120,14 @@ instance (Enum k, Show k) => Show (EnumSet k) where
 instance (Enum k, Read k) => Read (EnumSet k) where
   readPrec = parens . prec 10 $ do
     Ident "fromList" <- lexP
-    list <- readPrec
-    return (fromList list)
+    fromList <$> readPrec
+
+instance (ToJSON a) => ToJSON (EnumSet a) where
+    toJSON = toJSON . unWrap
```

Comment:
Does this use `ToJSON a`, or just encodes a list of integers?

Write tests, not only roundtrip properties, but also examples of what the encodings look like.

Reply:
Yes, it encode a list of integers at the end. 
I ll add tests.